### PR TITLE
Add phylum completion files

### PR DESCRIPTION
**Pull request by kylewillmon**
This patch adds shell completion files to the `phylum` formula.

After installing this new revision, the completion files will be included in the cellar and symlinks will be available at the following locations:

* `$(brew --prefix)/share/zsh/site-functions/_phylum`
* `$(brew --prefix)/etc/bash_completion.d/phylum.bash`
* `$(brew --prefix)/share/fish/vendor_completions.d/phylum.fish`

Additional details on shell completion in Homebrew can be found in [their docs](https://docs.brew.sh/Shell-Completion)

### CODE REVIEW

File: Formula/phylum.rb [hunk 1]
```diff
@@ -3,20 +3,19 @@ class Phylum < Formula
...
+  revision 1
   license "GPL-3.0-or-later"
   head "https://github.com/phylum-dev/cli.git", branch: "main"
 
-  bottle do
-    root_url "https://github.com/phylum-dev/homebrew-cli/releases/download/phylum-4.0.0"
-    sha256 cellar: :any_skip_relocation, arm64_monterey: "52545927b97f386d3afee0d7966d93bf9a244daa8b622a193f252deba315b60b"
-    sha256 cellar: :any_skip_relocation, monterey:       "02a934bcecf32b6a255d6f5d576d3fbe6cd1af282deda30d83137fbcfb72a729"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "df3bb96ca6bbc9542e1ef2e1002b845fd0dd4140aaeb76d261bfdf235c71de20"
-  end
```
Comment (cd-work) [inline Formula/phylum.rb:14]: Why was this removed?

Comment (kylewillmon) [reply Formula/phylum.rb:14]: I removed it for local testing. It will be replaced when the bottles are rebuilt by [this CI workflow](https://github.com/phylum-dev/homebrew-cli/blob/91d2144f260996dbd85dfc47452d338725d772b1/.github/workflows/publish.yml#L54-L58) once I get it to stop failing...

**Comment by kylewillmon** [comment]
It's worth noting that this PR updates the version of `phylum` to `4.0.0_1` even though the `4.0.1` release already exists upstream.

I did this primarily to test out formula revisions (i.e., improvements to the formula that are not related to an upstream release). I will bump the formula to `4.0.1` in a separate PR after this one is merged.

**Review by cd-work — APPROVED** [review]

**Review by cd-work — APPROVED** [review]
